### PR TITLE
[SIM_FLUTTER-52] [BE] Create Get Transactions API

### PR DESCRIPTION
**Pull request by IanUrriza10**
### Issue Links
- [backlog](https://framgiaph.backlog.com/view/SIM_FLUTTER-52)
- [slack](https://sun-philippine.slack.com/archives/C059L7XAXFB/p1687404203900159)
###  Notes
- Repopulate DB
    - first truncate transactions table
        - `php artisan db`
        -  `truncate transactions;` then `exit`
    - create transactions
        - `php artisan db:seed --class=TransactionSeeder` (takes time as there's 1 sec delay per insert)

- visit URLs
    - http://127.0.0.1:8000/api/transactions
    - http://127.0.0.1:8000/api/accounts/{account_id}
    - http://127.0.0.1:8000/api/users/{user_id}/accounts

- For Testing
    - Use `/login` first to get a `user_id` 
    - Use `Show accounts via user_id` to set the first `account_id`
    - All routes are now accessible in Transactions Collection
### Definition of done
- [x] create api/transactions route
- [x] display all transactions from the database
- [x] display all transactions of a specific account

### CODE REVIEW

**Comment by Dbulawan13** [comment]
<RESOLVED: NOT A BUG>
Comment:

- [ ] transaction cannot be fetch when sending via `{{api}}/accounts/{{account_id}}/transactions` since account_id is empty in global tab.

https://github.com/framgia/sph-flutter/assets/86758474/95957a30-b97b-437d-917e-56a3483d3c07

- [ ] The user_id is not the same when fetching the transaction via `{{api}}/transactions/{{transaction_id}}` maybe because some transaction_id is `NULL`

Via: `{{api}}/transactions/{{transaction_id}}`


`Compare to Login tab:`
image




Testing Hours: 0.50
Comments: 2

**Review by Dbulawan13 — DISMISSED** [review]
Approve PR #23 

Working:
- Able to Get transactions
   - via `ALL Accounts`
   - via `{{api}}/accounts/{{account_id}}/transactions `
   - via `{{api}}/transactions/{{transaction_id}}`





Testing Hours: 1
Comment: 0
Iteration: 0

File: backend/app/Http/Controllers/TransactionController.php [hunk 1]
```diff
@@ -2,7 +2,30 @@
...
+use App\Models\Account;
+use App\Models\Transaction;
+
 class TransactionController extends Controller
 {
-    //
+    public function index($account_id = null)
+    {
+
+        //For /accounts/{account_id}/transactions
```
Comment (jstephend-sun) [inline backend/app/Http/Controllers/TransactionController.php]:
```suggestion
    {

        // For /accounts/{account_id}/transactions
```

Comment (jstephend-sun) [inline backend/app/Http/Controllers/TransactionController.php]:
```suggestion
    {
        //For /accounts/{account_id}/transactions
```

File: backend/app/Http/Controllers/TransactionController.php [hunk 2]
```diff
@@ -2,7 +2,30 @@
...
+        if ($account_id) {
+            //For shallow nesting
```
Comment (jstephend-sun) [inline backend/app/Http/Controllers/TransactionController.php]:
```suggestion
            // For shallow nesting
```

File: backend/app/Http/Controllers/TransactionController.php [hunk 3]
```diff
@@ -2,7 +2,30 @@
...
+            // see https://laravel.com/docs/10.x/controllers#shallow-nesting
```
Comment (jstephend-sun) [inline backend/app/Http/Controllers/TransactionController.php]:
```suggestion
            // see https://laravel.com/docs/8.x/controllers#shallow-nesting
```

File: backend/app/Http/Controllers/TransactionController.php [hunk 4]
```diff
@@ -2,7 +2,30 @@
...
+            $account = Account::where('id', $account_id)->with(['accountTransactions'])->first();
+
+            return TransactionResource::collection($account->accountTransactions);
+        }
+
+        //For /transactions
```
Comment (jstephend-sun) [inline backend/app/Http/Controllers/TransactionController.php]:
```suggestion
        // For /transactions
```

**Review by jstephend-sun — CHANGES_REQUESTED** [review]
initial comments

**Review by jstephend-sun — APPROVED** [review]
Approved #23

review time: 5mins
comments: 1

**Pull request merged**